dask_expr/_merge.py

```diff
@@ -408,6 +402,35 @@ def _simplify_up(self, parent):
         return
 
 
+def create_assign_index_merge_transfer():
```

Comment:
I'm not sure I understand the benefit of this function. Is there a reason we cannot just use `assign_index_merge_transfer` directly (with the `pd` and `merge_transfer` imports moved into that function)?

Reply:
I don't want to do those imports on every function call, pandas isn't cheap for example.

It shouldn't matter, but I'd rather not take that chance. Also, I'd rather fail at graph construction time then during computation time if dependencies are messed up

Reply:
Okay, you only need to import `merge_transfer`, but I understand the interest in raising an error in `_layer`.

Can we just move this function to be a property of `merge_transfer` then?  For some reason, the `func = create_assign_index_merge_transfer()` pattern feels off to me. Sorry for nitpicking.

Reply:
This is a pretty common pattern in more functional environments 

I don't understand what you mean with property of merge_transfer

Reply:
>This is a pretty common pattern in more functional environments

Yeah I agree, but this just doesn't feel like a common pattern in dask/dask or dask-expr. I don't feel super strong about this, just trying to consider ways to keep style consistent throughout.

>I don't understand what you mean with property of merge_transfer

Oops! That was a typo. I was just saying that `create_assign_index_merge_transfer` could be a property of `HashJoinP2P`, but I don't love that pattern either (it's just a bit closer to what I'd expect).

I'll admit that I don't actually care much about this, so feel free to treat this comment thread as "non-blocking".